src/SmartSelect/Alignment.elm

```diff
@@ -1,46 +1,148 @@
-module SmartSelect.Alignment exposing (Alignment(..), containerClass, init, selectClass)
+module SmartSelect.Alignment exposing
+    ( Alignment, Params
+    , init, getElements
+    , style
+    , params
+    , view
+    )
 
-import Browser.Dom exposing (Element)
+{-| Determine the Alignment for the select options container
+
+
+# Definitions
+
+@docs Alignment, Params
+
+
+# Init
+
+@docs init, getElements
+
+
+# Style
+
+@docs style
+
+
+# Test
+
+@docs params
+
+-}
+
+import Browser.Dom as Dom exposing (Element)
+import Html exposing (Html, div)
+import Html.Attributes as Attrs
+import Task exposing (Task)
+import Task.Extra as TaskExtra
 
 
 type Alignment
+    = Alignment Position Placement
+
+
+type Position
+    = Position { x : Float, y : Float, width : Float }
+
+
+type Placement
     = Above
     | Below
 
 
-init : { container : Element, select : Element } -> Alignment
-init { container, select } =
+type Params
+    = Params { container : Element, select : Element, viewport : Dom.Viewport }
+
+
+init : Params -> Alignment
+init (Params { container, select, viewport }) =
     if
-        select.viewport.y
-            + select.element.y
+        select.element.y
             + select.element.height
             + container.element.height
-            >= select.viewport.height
+            >= viewport.viewport.height
     then
-        Above
+        Alignment
+            (Position
+                { x = select.element.x
+                , y = select.element.y - container.element.height
+                , width = select.element.width
+                }
+            )
+            Above
 
     else
-        Below
+        Alignment
+            (Position
+                { x = select.element.x
+                , y = select.element.y + select.element.height
+                , width = select.element.width
+                }
+            )
+            Below
+
+
+getElements : String -> String -> Task Dom.Error Params
+getElements containerId selectId =
+    Task.succeed (\container select viewport -> Params { container = container, select = select, viewport = viewport })
```

Comment:
Why are we mapping to `Params` here? Couldn't we map to `Alignment` directly?

Then we wouldn't need to expose and handle a `Params` type

Reply:
Yes, i will update it.